app/controllers/comments_controller.rb

```diff
@@ -141,7 +141,33 @@ def update
     authorize @comment
     if @comment.update(permitted_attributes(@comment).merge(edited_at: Time.zone.now))
       Mention.create_all(@comment)
-      redirect_to URI.parse(@comment.path).path, notice: "Comment was successfully updated."
+
+      # The following sets variables used in the index view. We render the
+      # index view directly to avoid having to redirect.
+      #
+      # Redirects lead to a race condition where we redirect to a cached view
+      # after updating data and we don't bust the cache fast enough before
+      # hitting the view, therefore stale content ends up being served from
+      # cache.
+      #
+      # https://github.com/forem/forem/issues/10338#issuecomment-693401481
+      @on_comments_page = true
+      @root_comment = @comment
+      @commentable = @comment.commentable
+      @commentable_type = @comment.commentable_type
+
+      case @commentable_type
+      when "PodcastEpisode"
+        @user = @commentable&.podcast
+      when "Article"
+        # user could be a user or an organization
+        @user = @commentable&.user
+        @article = @commentable
+      else
+        @user = @commentable&.user
+      end
```

Comment:
non-blocker: it might be nice to extract this into a method and out of the controller action itself.

Reply:
Agreed! I'm thinking how we do comments as a whole could use a refactor 🤔  I may circle back to clean up `index` and `update`.